core/src/main/java/org/apache/brooklyn/core/sensor/ReleaseableLatch.java

```diff
@@ -56,8 +60,11 @@
         }
 
         private static class MaxConcurrencyLatch implements ReleaseableLatch {
+            private static final Logger LOG = LoggerFactory.getLogger(MaxConcurrencyLatch.class);
+
             private int permits;
             private transient final Semaphore sem;
+            private transient final Set<Entity> ownerEntities = new ConcurrentHashSet<Entity>();
```

Comment:
Good spot, haven't considered this case. It's handled and covered by tests though - there's a `readResolve` further down. Added a comment.